src/ray/raylet/node_manager.cc

```diff
@@ -704,15 +752,7 @@ void NodeManager::ProcessDisconnectClientMessage(
     // If the worker was an actor, add it to the list of dead actors.
     const ActorID &actor_id = worker->GetActorId();
     if (!actor_id.is_nil()) {
-      // TODO(rkn): Consider broadcasting a message to all of the other
-      // node managers so that they can mark the actor as dead.
-      RAY_LOG(DEBUG) << "The actor with ID " << actor_id << " died.";
-      auto actor_entry = actor_registry_.find(actor_id);
-      RAY_CHECK(actor_entry != actor_registry_.end());
-      actor_entry->second.MarkDead();
-      // For dead actors, if there are remaining tasks for this actor, we
-      // should handle them.
-      CleanUpTasksForDeadActor(actor_id);
+      HandleDisconnectedActor(actor_id, true);
```

Comment:
One thing that's a little weird now is that there's going to be a period after the actor process initially fails where the node does not actually record that the actor is dead, since we wait until the GCS notification before updating the actor's state. I wonder if this will cause any issues, such as repeatedly trying and failing to assign a task to a dead actor? It would be nice to make the logic so that tasks submitted in the meantime get routed to the `MethodsWaitingForActorCreation` queue.

Also, we should make sure that the actor gets marked as `DEAD` instead of `BEING_RECONSTRUCTED` if it was intentionally killed. Probably, we should change the argument to this method from `push_warning` to something like `intentional_disconnect`, then propagate the value through to `HandleDisconnectedActor` to decide the actor's new state.

Reply:
1) I think we can avoid that issue by updating the local actor registry's state immediately when actor disconnects.
2) good catch, I forgot to change this.